[IDP Docs] Refactor Workflows (#9272)
harness/developer-hub: docs/internal-developer-portal/flows/overview.md

---
title: Self Service Workflows Overview
description: Get started with Self Service Workflows in Harness IDP. 
sidebar_position: 1
sidebar_label: Overview
redirect_from:
  - /docs/internal-developer-portal/features/service-onboarding-pipelines
---

**Service onboarding** in today’s tech world is slow and tedious. Developers often spend days—or even weeks—setting up new software and completing Day 2 operations. This inefficiency arises from either waiting for ticket resolutions (TicketOps) or manually handling repetitive tasks, which results in a poor developer experience and decreased productivity.

Harness IDP addresses these challenges with **Self-Service Workflows**.

Workflows enable developer self-service by automating manual tasks and processes. Using Workflows, platform engineering teams can:
- Automate new service onboarding.

> - Simplify Day 2 operations for developers.

Reviewer: Let's mention some examples of Day 2 operations. Refer to the product screenshot we share on overview pages.

- Provide developers with **golden paths** to production that include guardrails and best practices.

## Workflow Basics 
Let’s get started with the fundamental steps to **understand, build, and configure a Workflow** from scratch.  

### [Get Started with Workflows](/docs/internal-developer-portal/get-started/workflow-quickstart.md)
Ready to start using Workflows? Follow our [**Quickstart Guide**](/docs/internal-developer-portal/get-started/workflow-quickstart.md) and **create your first Workflow** in just a few minutes.

1. **Register the Workflow** using the `workflow.yaml` file configuration.  
2. **Execute the Workflow** by providing input details as defined in the configuration.  
3. **Inputs are processed**, **backend actions are triggered**, and **orchestration pipelines** execute tasks according to the Workflow logic.  
4. **Outputs are generated** as specified in the `workflow.yaml` configuration.  

> ### [Understanding Workflow.yaml](/docs/internal-developer-portal/flows/worflowyaml.md) 

Reviewer: ```suggestion
### [Understanding Workflow YAML](/docs/internal-developer-portal/flows/worflowyaml.md) 
```

Workflow is defined using a **YAML configuration** file, typically named **`workflow.yaml`**, which contains all the metadata required for the Workflow. This file is stored in the **root directory** of the source code repository in your connected Git provider.

#### **Components of `Workflow.yaml`**
The `workflow.yaml` file is divided into three key components:

1. **Frontend**: Defines the **input fields** required for the Workflow.  
2. **Backend**: Configures the **actions** to be triggered and the **orchestration pipelines** to be executed during the Workflow.  

> 3. **Outputs**: Specifies the **output variables** to be used after backend execution.  

Reviewer: ```suggestion
3. **Outputs**: Specifies the **output variables** to be shown to developers after the execution.  
```

> Learn more about the [**Workflow YAML**](/docs/internal-developer-portal/flows/worflowyaml.md) here. 

Reviewer: ```suggestion
Learn more about the [**Workflow YAML syntax**](/docs/internal-developer-portal/flows/worflowyaml.md) here. 
```
